proxy/src/main/java/com/wavefront/agent/DataDogAgentHandler.java

```diff
@@ -0,0 +1,409 @@
+package com.wavefront.agent;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpContent;
+import io.netty.handler.codec.http.HttpHeaders;
+import io.netty.handler.codec.http.HttpObject;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.codec.http.LastHttpContent;
+import io.netty.util.CharsetUtil;
+
+import static io.netty.handler.codec.http.HttpHeaders.Names.*;
+
+import java.nio.ByteBuffer;
+import java.util.zip.Inflater;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.UUID;
+
+import com.wavefront.agent.api.ForceQueueEnabledAgentAPI;
+import sunnylabs.report.ReportPoint;
+
+import org.codehaus.jackson.map.ObjectMapper;
+import org.codehaus.jackson.JsonNode;
+
+/**
+ * This class is a netty channel handler for metrics arriving from
+ * a DataDog agent.  This handler operates as a mini HTTP server and returns
+ * a 200 status code for all requests (unless an exception occurs).
+ * To use this, change datadog agent configuration(s) to point the dd_url
+ * value to the Wavefront proxy where this handler is running.
+ * To use this to send data to both datadog and WF, be sure to put a proxy
+ * in front of this handler (teeproxy, or similar) and duplicate requests to
+ * this handler.
+ * This is based off the netty example HttpSnoopServer found here:
+ * https://github.com/netty/netty/tree/4.1/example/src/main/java/io/netty/example/http/snoop
+ * This class was created from the example provided in HttpSnoopServerHandler
+ * class in the above directory.
+ */
+@ChannelHandler.Sharable
+public class DataDogAgentHandler extends SimpleChannelInboundHandler<Object> {
```

Comment:
we already have a jetty json endpoint that does all the parsing with jersey. Can you take a look at that instead? Not that your implementation in netty is deficient but if there's already something working, it seems like it would be a lot less code to have another HTTP endpoint that takes a JSON object (you can probably even POJO-tise it)